packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -462,7 +461,8 @@ export const RemoteServerFactory = {
             locals.requestStart = Date.now()
             locals.afterResponseCalled = false
             locals.responseCaching = {}
-            locals.requestId = _nextRequestId++
+            locals.requestId = req.headers['x-amzn-requestid'] || uuidv4()
```

Comment:
@alexvuong I'm guessing that the reason you are reassigning the requestId here is because we'll get all the logging changes for free since `localDevLog` is called with it for the most part?

Reply:
Not really, I wanted to make sure that requestId is already defined since we will pass it as a default value to CorrelationIdProvider in server-side rendering pipeline, and also to respect the currently implementation that it is always defined (as _nextRequestId++)

However, I think a better way to do it is to separate how to set up requesId different in devServer and remoteServer. 

On RemoteServer, we know that the req.headers['x-amzn-requestid'] is most likely to be defined. On DevServer, we need to generate uuid because we don't have that value. It makes sense to have the uuid implementation in pwa-kit-dev for that purpose.